features/storage/bd/MemBlockDevice.cpp

```diff
@@ -0,0 +1,114 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "MemBlockDevice.h"
+
+
+MemBlockDevice::MemBlockDevice(bd_count_t count, bd_size_t size)
+    : _count(count)
+    , _size(size)
+    , _blocks(0)
+{
+}
+
+MemBlockDevice::~MemBlockDevice()
+{
+    if (_blocks) {
+        for (bd_count_t i = 0; i < _count; i++) {
+            free(_blocks[i]);
+        }
+
+        free(_blocks);
+        _blocks = 0;
+    }
+}
+
+bd_error_t MemBlockDevice::init()
+{
+    if (!_blocks) {
+        _blocks = (uint8_t**)malloc(_count*sizeof(uint8_t*));
```

Comment:
Can update to new[], this failure is unlikely relative to the allocation of the actual blocks of memory though it may still fail as a result of the user requesting too large a block of memory.